src/main/java/org/terasology/launcher/updater/LauncherUpdaterFactory.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.updater;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.launcher.util.DirectoryUtils;
+import org.terasology.launcher.util.FileUtils;
+import org.terasology.launcher.version.TerasologyLauncherVersionInfo;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+/**
+ * Use this class to get the appropriate platform-specific updater for the launcher.
+ */
+public final class LauncherUpdaterFactory {
+
+    private enum UpdateStrategy {
+        SELF,
+        WIN_INSTALLER
+    }
+
+    private static final Logger logger = LoggerFactory.getLogger(AbstractLauncherUpdater.class);
+
+    private LauncherUpdaterFactory() {
+    }
+
+    /**
+     * @return The right updater for the running instance of the launcher.
+     * It is detected by reading the updateStrategy file in the launcher installation directory.
```

Comment:
Should we keep _updateStrategy_ as separate file or put it as option in the launcher settings somehow? Would it make sense to change it after packaging? 
(I guess the current solution is fine, just wanted to ask)